Assets/Scripts/Gameplay/ConnectionManagement/ConnectionState/ClientConnectingState.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using Unity.Multiplayer.Samples.BossRoom.Shared.Infrastructure;
+using Unity.Multiplayer.Samples.Utilities;
+
+namespace Unity.Multiplayer.Samples.BossRoom
+{
+    /// <summary>
+    /// Connection state corresponding to when a client is attempting to connect to a server. If successful,transitions
+    /// to the Connected state. If not, transitions to the Offline state.
+    /// </summary>
+    public class ClientConnectingState : ConnectionState
+    {
+        IPublisher<ConnectStatus> m_ConnectStatusPublisher;
```

Comment:
I see this in a good majority of your states, why not put it in the base class? It'd make sense to have every state be able to publish connection status.

Reply:
With our current DI implementation, I would have to inject this in every daughter class anyways, but once VContainer is merged I'll update this

Reply:
So not sure I understand, if this was in the base class, you'd still need to inject sure, but you still wouldn't need to add this publisher to all your state, that's my point. It'd make sense to have all states have a publisher. The Inject attribute is inherited, so this should work in vcontainer.